### PR TITLE
fix tooltip with overlapping points

### LINKED ISSUE #221: Overlapping tooltip selects wrong data even if element hidden.
Problem: Currently if there are two overlapping points, the first tooltip will always be selected even if it is hidden.  
In tooltip.js, the function show is: 

``` javascript
var show = function (d) {
    clearHideTimer.call(this);

    var dataPoints = findOriginalDataPoint(d);

   /**line 140**/ this.tooltipElement.select('.text').html(getTooltipText.call(this, dataPoints[0] || d, dataPoints));

    var pos = positionTooltip.call(this, d);

    this.tooltipElement
        .style('top', pos.y + 'px')
        .style('left', pos.x + 'px');

    changeOpacity.call(this, this.options.tooltip.opacity, this.options.tooltip.showTime);
};
```

I believe that on line 140, dataPoints[0] || d is overriding the data and only picking the first in the set since || will always pick the first point: 

``` javascript
this.tooltipElement.select('.text').html(getTooltipText.call(this, dataPoints[0] || d, dataPoints));
```

Switching the order of the || works for the project I'm working on with overlapping points.

``` javascript
this.tooltipElement.select('.text').html(getTooltipText.call(this, d || dataPoints[0], dataPoints));
```

However, I am wondering what findOriginalDatPoint(d) is suppose to do. It seems to select all data with  the same x and y as the parameter's and then the html text is based on the first one (dataPoints[0]). Can someone confirm if this is the case and if switching the order makes sense?

``` javascript
function findOriginalDataPoint(d) {
    var res = [];
    _.each(data, function(series, seriesIndex) {
        var name = series.name;
        _.each(series.data, function(point) {
            if (point.x === d.x && d.y === point.y) {
                res.push(_.extend(point, {
                    series: name,
                    seriesIndex: seriesIndex
                }));
            }
        });
    });
    return res;
}
```

### PR DESCRIPTION
**Pull request by senluchen2015**
I believe this will resolve #221

### CODE REVIEW

File: src/scripts/visualizations/tooltip.js [hunk 1]
```diff
@@ -137,7 +137,7 @@
 
             var dataPoints = findOriginalDataPoint(d);
 
-            this.tooltipElement.select('.text').html(getTooltipText.call(this, dataPoints[0] || d, dataPoints));
+            this.tooltipElement.select('.text').html(getTooltipText.call(this, d || dataPoints[0], dataPoints));
```
Comment (drosen0) [inline src/scripts/visualizations/tooltip.js]: I believe the intent is for the original data point, if found, to override the data point passed in. How about this:
```js
this.tooltipElement.select('.text').html(getTooltipText.call(this, _.isArray(dataPoints) && dataPoints[0] || d, dataPoints));
```

Comment (drosen0) [inline src/scripts/visualizations/tooltip.js]: Never mind, after reading #221, I realize I missed the point.

**Comment by mmrj** [comment]
@senluchen2015: @drosen0 closed this without merging because related issue #221 was closed as not a bug.

**Comment by drosen0** [comment]
Reopening because Contour is calling the custom formatter with the wrong data point.

**Pull request merged**